relay-sampling/src/lib.rs

```diff
@@ -84,1522 +84,39 @@ pub use dsc::DynamicSamplingContext;
-                    start: Some(Utc.with_ymd_and_hms(1970, 10, 10, 0, 0, 0).unwrap()),
-                    end: Some(Utc.with_ymd_and_hms(1970, 10, 12, 0, 0, 0).unwrap()),
-                },
-                decaying_fn: DecayingFunction::Linear { decayed_value: 0.3 },
-            },
-            SamplingRule {
-                condition: and(vec![]),
-                sampling_value: SamplingValue::SampleRate { value: 0.02 },
-                ty: RuleType::Trace,
-                id: RuleId(3),
-                time_range: Default::default(),
-                decaying_fn: Default::default(),
-            },
-        ]);
-
-        // factor match first rule and early return third rule
-        let event = mocked_event(EventType::Transaction, "transaction", "1.1.1", "testing");
-        let dsc = mocked_dynamic_sampling_context(
-            "root_transaction",
-            "1.1.1",
-            "dev",
-            "vip",
-            "user-id",
-            None,
-        );
-        // We will use a time in the middle of 10th and 11th.
-        let result = match_against_rules(
-            &config,
-            &event,
-            &dsc,
-            Utc.with_ymd_and_hms(1970, 10, 11, 0, 0, 0).unwrap(),
-        );
-        assert_trace_match!(result, 0.03, dsc, 1, 3);
-
-        // early return second rule
-        let event = mocked_event(EventType::Transaction, "transaction", "1.1.1", "testing");
-        let dsc = mocked_dynamic_sampling_context(
-            "root_transaction",
-            "1.1.1",
-            "prod",
-            "vip",
-            "user-id",
-            None,
-        );
-        // We will use a time in the middle of 10th and 11th.
-        let result = match_against_rules(
-            &config,
-            &event,
-            &dsc,
-            Utc.with_ymd_and_hms(1970, 10, 11, 0, 0, 0).unwrap(),
-        );
-        assert!(matches!(result, Some(SamplingMatch { .. })));
-        if let Some(spec) = result {
-            assert!(
-                (spec.sample_rate - 0.45).abs() < f64::EPSILON, // 0.45
-                "did not use the sample rate of the second rule"
-            )
-        }
-
-        // early return third rule
-        let event = mocked_event(EventType::Transaction, "transaction", "1.1.1", "testing");
-        let dsc = mocked_dynamic_sampling_context(
-            "root_transaction",
-            "1.1.1",
-            "testing",
-            "vip",
-            "user-id",
-            None,
-        );
-        // We will use a time in the middle of 10th and 11th.
-        let result = match_against_rules(
-            &config,
-            &event,
-            &dsc,
-            Utc.with_ymd_and_hms(1970, 10, 11, 0, 0, 0).unwrap(),
-        );
-        assert_trace_match!(result, 0.02, dsc, 3);
-    }
-
-    #[test]
-    /// test that the correct match is performed when replay id is present in the dsc.
-    fn test_sampling_match_with_trace_replay_id() {
```

Comment:
Removed because it's testing behaviour to a specific thing that implements `Getter` which is not the domain of this crate.